src/composables/protocol.ts

```diff
@@ -0,0 +1,56 @@
+import type {
+  Class,
+  IDefaultComposableOptions,
+  Protocol,
+} from '@/types';
+import { BaseProtocol } from '@/protocols/BaseProtocol';
+import { AeternityProtocol } from '@/protocols/aeternity/libs/AeternityProtocol';
+import { BitcoinProtocol } from '@/protocols/bitcoin/libs/BitcoinProtocol';
+import {
+  PROTOCOL_AETERNITY,
+  PROTOCOL_BITCOIN,
+} from '@/constants';
+import { watchUntilTruthy } from '@/utils';
+import { useAccounts } from './accounts';
+
+let isProtocolRegistryReady = true;
+
+const protocolClassReferences: Record<Protocol, Class<BaseProtocol> > = {
+  [PROTOCOL_AETERNITY]: AeternityProtocol,
+  [PROTOCOL_BITCOIN]: BitcoinProtocol,
```

Comment:
`Class<BaseProtocol>` looks too meta, maybe just
```suggestion
const protocolClassReferences: Record<Protocol, () => BaseProtocol> = {
  [PROTOCOL_AETERNITY]: () => new AeternityProtocol(),
  [PROTOCOL_BITCOIN]: () => new BitcoinProtocol(),
```

Reply:
Nice. Thanks.

Reply:
I don't think the change suggested by Denys is good for us. Storing the class instead of the callback that returns new instance gives us possibility to access the static properties of the class.

Reply:
Keeping it according to Bart's suggestions.